AmethystAPI/CMakeLists.txt

```diff
@@ -10,26 +10,117 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CONFIGURATION_TYPES "RelWithDebInfo" CACHE STRING "Build configurations" FORCE)
 set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build, options are: RelWithDebInfo" FORCE)
 
+# Fix STL structures on debug mode
+add_compile_definitions(_ITERATOR_DEBUG_LEVEL=0)
```

Comment:
To fully address the runtime library incompatibility, the MSVC_RUNTIME_LIBRARY property should be set instead. While iterator debug level works for fixing some issues related to container sizes, it does not resolve the mismatch in debug/release allocator.